lisa/base_tools/uname.py

```diff
@@ -3,15 +3,19 @@
 
 import re
 from dataclasses import dataclass
+from semver import VersionInfo
 
 from lisa.executable import Tool
 from lisa.util import LisaException
 
+DEFAULT_VERSION = VersionInfo(0, 0, 0)
+
 
 @dataclass
 class UnameResult:
     has_result: bool
     kernel_version: str = ""
+    kernel_version_parsed: VersionInfo = DEFAULT_VERSION
```

Comment:
1. It doesn't need a default value. Move it up to kernel_version. All required arguments should be above optional.
2. The name should be `kernel_version`, and the str version should be like `raw_kernel_version` or something else. Because we encourage people to use structured version.

Reply:
Regarding 2, currently `kernel_version: str ` is used primarily for logging purpose in the codebase. Also, `VersionInfo` is capable of dumping the raw string in formatted logs. 

```
>>> ver = VersionInfo.parse("5.8.0-1039-azure")
>>> ver
VersionInfo(major=5, minor=8, patch=0, prerelease='1039-azure', build=None)
>>> f"{ver}"
'5.8.0-1039-azure'
```

The additional raw string variable will be unused. Wouldn't it be cleaner to leave raw string variable and work with `VersionInfo` class?

Reply:
Let's keep the raw string. I haven't checked how the `__str__` is implemented in `VersionInfo`. It's safe to maintain the raw string by us.

Reply:
I can keep it if you are insisting. But the implementation seems okay to me : 

https://github.com/python-semver/python-semver/blob/e93b6def40194f374623ce5e524a3dcfa40f7682/src/semver/version.py#L486-L492

Reply:
It's not the raw string, it's a standard string. If some version are not following this pattern, and used other format, our code can parse it into VersionInfo, but cannot get it back. 

Check the VersionInfo parser on distro version. We may need to reuse the logic in kernel version, if we found some version  cannot be parsed by default patterns.

https://github.com/microsoft/lisa/blob/main/lisa/operating_system.py#L237

Reply:
Agreed! It makes sense to reuse the logic from `operating_system.py`. How about we can move this logic into utils?

Reply:
Yes, it can be.